core/src/wallet/tezos/explorers/ExternalTezosLikeBlockchainExplorer.cpp

```diff
@@ -83,21 +98,47 @@ namespace ledger {
-                        // Sometimes network is sending 0 for fees
-                        if (fees == "0") {
-                            fees = api::TezosConfigurationDefaults::TEZOS_DEFAULT_FEES;
-                        } else if (fees.find('.') != std::string::npos) {
-                            fees = api::BigInt::fromDecimalString(fees, 6, ".")->toString(10);
+
+                        // Return 0 if the block had no transaction at all
+                        if (!json.HasMember(txCountField) || !json[txCountField].IsString()) {
+                            return std::make_shared<BigInt>(0);
+                        }
+
+                        const auto totalTx = api::BigInt::fromIntegerString(json[txCountField].GetString(), 10);
+                        const auto totalFees = api::BigInt::fromDecimalString(json[feesField].GetString(), 6, ".");
+                        std::string fees = api::TezosConfigurationDefaults::TEZOS_DEFAULT_FEES;
+                        if (fees != "0" && totalTx->intValue() != 0) {
+                            fees = totalFees->divide(totalTx)->toString(10);
                         }
-                        // Since nodes are giving some awkward values, we set a threshold to avoid having really fees
-                        // Factor for threshold is inspired from other XTZ wallets
+                        // Since nodes are giving some awkward values, we set a threshold to avoid
+                        // having really fees Factor for threshold is inspired from other XTZ
+                        // wallets
                         return std::make_shared<BigInt>(std::min(std::stoi(fees), std::stoi(api::TezosConfigurationDefaults::TEZOS_DEFAULT_MAX_FEES)));
                     });
         }
 
+        Future<std::shared_ptr<BigInt>>
+        ExternalTezosLikeBlockchainExplorer::getGasPrice() {
+            const bool parseNumbersAsString = true;
+            const auto gasPriceField = "gas_price";
+
+            return _http->GET("block/head")
+                    .json(parseNumbersAsString).mapPtr<BigInt>(getContext(), [=](const HttpRequest::JsonResult &result) {
+                        auto &json = *std::get<1>(result);
+
+                        if (!json.IsObject() || !json.HasMember(gasPriceField) ||
+                            !json[gasPriceField].IsString()) {
+                            throw make_exception(api::ErrorCode::HTTP_ERROR,
+                                                 fmt::format("Failed to get gas_price from network, no (or malformed) field \"{}\" in response", gasPriceField));
+                        }
+                        const std::string apiGasPrice = json[gasPriceField].GetString();
+                        const std::string picoTezGasPrice = api::BigInt::fromDecimalString(apiGasPrice, 6, ".")->toString(10);
+                        return std::make_shared<BigInt>(std::stoi(picoTezGasPrice));
```

Comment:
Why not just return the result of `api::BigInt::fromDecimalString` here?

Reply:
because when I use BigInt->toInt64 later I don't know if I get `500` or `50` for a value like `"0.50"`